pkg/kubecost/allocationprops.go

```diff
@@ -439,7 +440,22 @@ func (p *AllocationProperties) Intersection(that *AllocationProperties) *Allocat
 		intersectionProps.ControllerKind = p.ControllerKind
 	}
 	if p.Namespace == that.Namespace {
+
 		intersectionProps.Namespace = p.Namespace
+		// ignore the incoming labels from unallocated or unmounted special case pods
+		if p.AggregatedMetadata || that.AggregatedMetadata {
+			intersectionProps.AggregatedMetadata = true
+			if p.Container == UnmountedSuffix || p.Container == UnallocatedSuffix {
+				intersectionProps.Annotations = that.Annotations
+				intersectionProps.Labels = that.Labels
+			} else if that.Container == UnmountedSuffix || that.Container == UnallocatedSuffix {
+				intersectionProps.Annotations = p.Annotations
+				intersectionProps.Labels = p.Labels
```

Comment:
My first thought would be that annotations and labels would be dropped entirely if we're dealing with an unmounted or unallocated Allocation. This looks like we're overriding the annotations/labels instead.

I'm certainly missing context here, so you may have already considered this.

Reply:
Yes, this logic looks weird and probably deserves a comment with what I am about to share: 

The core change is that we now maintain the intersection of the labels/annotations of the two AllocationProperties objects that we add. Normally, this works fine, however, our special case unallocated/unmounted Allocations are built by us and never have any labels or annotations. Because of this, as these get aggregated up (by namespace or otherwise) they have the effect of always clearing out the intersection, regardless if all the other actual pods/etc have them. 

This logic is designed to effectively ignore the unmounted/unallocated objects and just copy over the labels from the other object - we only take the intersection of 'legitimate' allocations.